### ISSUE TITLE
Future of this project

### ISSUE THREAD

**Issue by FranklinYu** [flackr/circ#438]
Now that Chrome App [will be gone in June 2021 even for Chrome OS](https://blog.chromium.org/2020/01/moving-forward-from-chrome-apps.html) (for individual users), is there any plan to move forward? (Or is it even possible to do so?)

**Comment by scoopgracie** [comment]
WITW? Chrome apps are the main way software is written for Chrome OS!!!

**Comment by kj7rrv** [comment]
Anyway, I think a PWA + a Chrome _extension_ might be a good idea.

**Comment by kj7rrv** [comment]
Maybe make the extension client-agnostic so other Web apps that need IRC access can use it.

**Comment by flackr** [comment]
I don't believe there is a [chrome extension API](https://developer.chrome.com/extensions/api_index) for [making the raw TCP connections](https://developer.chrome.com/apps/sockets_tcp) necessary for connecting to an IRC server, but please correct me if I am mistaken.

There is a sample PWA that I've started on the server branch but it will require running a server (currently NodeJS) on some computer to make the real IRC connections.

**Comment by kj7rrv** [comment]
Can't extensions use all the APIs apps can?

**Comment by kj7rrv** [comment]
I was even thinking of a general extension to work for any web app that
wants a socket connection.

**Comment by FranklinYu** [comment]
> Can't extensions use all the APIs apps can?

No, extensions have a much more restricted API. Socket API is only one of them.

**Comment by kj7rrv** [comment]
Weird. How do the SSH extensions work? (There are non-app ones.)

**Comment by FranklinYu** [comment]
Any example?

**Comment by kj7rrv** [comment]
https://chrome.google.com/webstore/detail/secure-shell/iodihamcpbpeioajjeobimgagajmlibd

**Comment by FranklinYu** [comment]
That’s the only whitelisted one. I mean, since it’s made by the Chrome team, it’s very easy for them to whitelist their own extension, essentially making the API a private one. CIRC in contrast doesn’t get the bless from them, so it’s more difficult to rely on their private API (technically possible by looking at their code but that’s reverse-engineering because the API would be undocumented).

**Comment by kj7rrv** [comment]
Maybe we should wait a while? Maybe Google will startallowing sockets for
extensions.

**Comment by FranklinYu** [comment]
> Maybe Google will start allowing sockets for extensions.

Yes, that would possibly be the best scenario.

**Comment by kj7rrv** [comment]
I doubt they would permanently block the entire platform from something as
important as sockets.

**Comment by FranklinYu** [comment]
I think I missed part of the announcement. Chrome App on Chrome OS will be until the middle of next year, but users with Chrome Enterprise or Chrome Education (covering a large part of Chrome OS users) will have one more year. At least this buys more time for students getting Chromebook from schools/universities, and we have adequate time to consider the way out.

I’m editing the issue description for that.

**Comment by FranklinYu** [comment]
The [raw socket proposal](https://github.com/WICG/raw-sockets) is under early discussion. If this proposal is approved, then it becomes possible to host a static page (for example, on GitHub Pages) as an IRC client.
